types/views/views.go

```diff
@@ -88,6 +88,36 @@ func (v SliceView[T, V]) AsSlice() []V {
 	return v.AppendTo(nil)
 }
 
+func (v SliceView[T, V]) Iterator() SliceViewIterator[T, V] {
+	return SliceViewIterator[T, V]{}
+}
+
+// SliceViewIterator provides an iterator interface over a SliceView.
+//
+// A typical usage example is
+//   var s Slice[T]
+//   for i := s.Iterator(); i.Valid(); i = i.Next() {
+//     v := i.Val()
+//   }
+type SliceViewIterator[T ViewCloner[T, V], V StructView[T]] struct {
+	s SliceView[T, V]
+	i int
+}
+
+// Next increments the iterator.
+func (si SliceViewIterator[T, V]) Next() SliceViewIterator[T, V] {
```

Comment:
I'm confused. This modifies a local copy and returns it.

But above in the `local.go` example, `i.Next()` is called without assigning the output to anything.

Should this be declared on a pointer receiver instead?
```go
func (si *SliceViewIterator[T, V]) Next()
```

Given that this is a concrete type, I suspect that the compiler will prove that usages do not escape and we don't have any allocations.